src/contracts/libraries/BeaconChainProofs.sol

```diff
@@ -414,4 +406,100 @@ library BeaconChainProofs {
         return sha256(abi.encodePacked(validatorPubkey, bytes16(0)));
     }
+
+    /**
+     * @notice Parses a balanceRoot to get the uint64 balance of a validator.  
+     * @dev During merkleization of the beacon state balance tree, four uint64 values are treated as a single 
+     * leaf in the merkle tree. We use validatorIndex % 4 to determine which of the four uint64 values to 
+     * extract from the balanceRoot.
+     * @param balanceRoot is the combination of 4 validator balances being proven for
+     * @param validatorIndex is the index of the validator being proven for
+     * @return The validator's balance, in Gwei
+     */
+    function getBalanceAtIndex(bytes32 balanceRoot, uint40 validatorIndex) internal pure returns (uint64) {
+        uint256 bitShiftAmount = (validatorIndex % 4) * 64;
+        return 
+            Endian.fromLittleEndianUint64(bytes32((uint256(balanceRoot) << bitShiftAmount)));
+    }
+
+    /**
+     * @dev Retrieve the withdrawal timestamp
+     */
+    function getWithdrawalTimestamp(WithdrawalProof memory withdrawalProof) internal pure returns (uint64) {
+        return
+            Endian.fromLittleEndianUint64(withdrawalProof.timestampRoot);
+    }
+
+    /**
+     * @dev Converts the withdrawal's slot to an epoch
+     */
+    function getWithdrawalEpoch(WithdrawalProof memory withdrawalProof) internal pure returns (uint64) {
+        return
+            Endian.fromLittleEndianUint64(withdrawalProof.slotRoot) / SLOTS_PER_EPOCH;
+    }
+
+    /**
+     * Getters for validator fields:
+     * 0: pubkey
+     * 1: withdrawal credentials
+     * 2: effective balance
+     * 3: slashed?
+     * 4: activation elligibility epoch
+     * 5: activation epoch
+     * 6: exit epoch
+     * 7: withdrawable epoch
+     */
+
+    /**
+     * @dev Retrieves a validator's pubkey hash
+     */
+     function getPubkeyHash(bytes32[] memory validatorFields) internal pure returns (bytes32) {
+        return 
+            validatorFields[VALIDATOR_PUBKEY_INDEX];
+    }
+
+    function getWithdrawalCredentials(bytes32[] memory validatorFields) internal pure returns (bytes32) {
+        return
+            validatorFields[VALIDATOR_WITHDRAWAL_CREDENTIALS_INDEX];
+    }
+
+    /**
+     * @dev Retrieves a validator's effective balance (in gwei)
+     */
+     function getEffectiveBalanceGwei(bytes32[] memory validatorFields) internal pure returns (uint64) {
+        return 
+            Endian.fromLittleEndianUint64(validatorFields[VALIDATOR_BALANCE_INDEX]);
+    }
+
+    /**
+     * @dev Retrieves a validator's withdrawable epoch
+     */
+    function getWithdrawableEpoch(bytes32[] memory validatorFields) internal pure returns (uint64) {
+        return 
+            Endian.fromLittleEndianUint64(validatorFields[VALIDATOR_WITHDRAWABLE_EPOCH_INDEX]);
+    }
+
+    /**
+     * Getters for withdrawal fields:
+     * 0: withdrawal index
+     * 1: validator index
+     * 2: execution address
+     * 3: withdrawal amount
```

Comment:
likewise, not seeing getters for `withdrawal index` or  `execution address`

Reply:
probably not neessary to add getters we dont use?

Reply:
> probably not neessary to add getters we dont use?

yeah, I agree with that, just noting the seeming inconsistency with the comment here

Reply:
I think it's fine as-is, overall.